c/flake.nix

```diff
@@ -0,0 +1,44 @@
+{
+  description = "Aux template for C project";
+
+  inputs.nixpkgs.url = "github:auxolotl/nixpkgs/nixos-unstable";
+
+  outputs =
+    { self, nixpkgs }:
+    let
+      forAllSystems =
+        function:
+        nixpkgs.lib.genAttrs [
+          "x86_64-linux"
+          "aarch64-linux"
+          "x86_64-darwin"
+          "aarch64-darwin"
+          "i686-linux"
+          "mipsel-linux"
+          "powerpc64le-linux"
+        ] (system: function nixpkgs.legacyPackages.${system});
+    in
+    rec {
+      devShells = forAllSystems (pkgs: {
+        default = pkgs.mkShell {
+          hardeningDisable = [ "fortify" ];
+          inputsFrom = pkgs.lib.attrsets.attrValues packages;
```

Comment:
```suggestion
          inputsFrom = [ packages.${pkgs.system}.hello ];
```

Reply:
To me it would make more sense for this to be dynamic, because i cant think of a scenario where you wouldn't add the build inputs of a package within the dev shell

Reply:
> because i cant think of a scenario where you wouldn't add the build inputs of a package within the dev shell

slightly different builds of the same package. this is done in [prismlauncher](https://github.com/prismlauncher/prismlauncher) for example

this also duplicates inputs due to the `default` package (which only really slows eval time a bit, but still) and currently doesn't work at all

`lib.attrValues packages` evaluates to `[ { default = { ... }; hello = { ... }; } { default = { ... }; hello = { ... }; } { ... } ]`